velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -379,7 +419,16 @@ void HiveDataSink::appendData(RowVectorPtr input) {
   // must be zero.
   if (!isBucketed() && partitionIdGenerator_->numPartitions() == 1) {
     const auto index = ensureWriter(HiveWriterId{0});
-    write(index, input);
+    auto dataInput = input;
+    if (insertTableHandle_->tableStorageFormat() ==
```

Comment:
This code repeats. Let's refactor to avoid copy-paste.

I suggest the following:

- In the constructor, figure out which input columns need to be written and store indices of these in a member variable.
- Here, right after "Lazy load all the input columns." loop, create a vector with just the necessary columns from input. Use this vector in calls to write.

Reply:
@mbasmanova Good suggestions. I have updated.